pallets/automation-time/src/tests.rs

```diff
@@ -209,16 +229,35 @@ fn cancel_must_be_owner() {
 fn cancel_task_must_exist() {
 	new_test_ext().execute_with(|| {
 		let scheduled_time = SCHEDULED_TIME + 300;
-		let task_id = schedule_task(ALICE, vec![40], scheduled_time, vec![2, 4, 5]);
+		let task = Task::<Test>::create_event_task(ALICE, vec![40], scheduled_time, vec![2, 4, 5]);
+		let task_id = BlakeTwo256::hash_of(&task);
 
-		assert_ok!(AutomationTime::cancel_task(Origin::signed(ALICE), task_id,));
 		assert_noop!(
 			AutomationTime::cancel_task(Origin::signed(ALICE), task_id),
 			Error::<Test>::TaskDoesNotExist,
 		);
 	})
 }
 
+#[test]
+fn cancel_task_not_found() {
```

Comment:
I don't get what this does.

Reply:
there is a weird edge case were somehow the task exists but isn't scheduled or in the overflow. There is no way this state should ever happen but if it does I want to know about it